misc/windows-deploy/Install-ECSAgent.ps1

```diff
@@ -0,0 +1,139 @@
+﻿# Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the
+# License is located at
+#
+#	http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+[cmdletbinding()]
+Param (
+    [Parameter(Mandatory=$false)]
+    [ValidateNotNullOrEmpty()]
+    [string]$Path = "$($PSScriptRoot)",
+
+    [Parameter(Mandatory=$false)]
+    [ValidateSet("Manual","AutomaticDelayedStart","Disabled")]
+    [string]$StartupType = 'Manual'
+)
+Begin {
+    $Script:ServiceName = "AmazonECS"
+    if (Get-Service | ?{$_.Name -eq "$($Script:ServiceName)"}) {
+        Write-Host "The $($Script:ServiceName) service already exists."
+        return
+    }
+    Function Initialize-Directory {
+        Param (
+            [string]$Path,
+            [string]$Name
+        )
+        [string]$targetDir = Join-Path "$($Path)" "$($Name)"
+        if (-Not (Test-Path $targetDir -ErrorAction:Ignore)) {
+            Write-Verbose "creating directory: $($targetDir)"
+            New-Item -Path $($targetDir) -ItemType:Directory -Force -ErrorAction:Continue | Out-Null
+        } else {
+            Write-Verbose "directory found: $($targetDir)"
+        }
+        return $targetDir
+    }
+
+    #
+    # Setup Default Directories
+    #
+    Write-Verbose "Setting up default directories"
+    # C:\ProgramData\Amazon
+    [string]$Script:AmazonProgramData = Initialize-Directory -Path $ENV:ProgramData -Name "Amazon"
+    # C:\ProgramData\Amazon\ECS
+    [string]$Script:ECSProgramData = Initialize-Directory -Path $Script:AmazonProgramData -Name  "ECS"
+    # C:\ProgramData\Amazon\ECS\data
+    [string]$Script:ECSData = Initialize-Directory -Path $Script:ECSProgramData -Name  "data"
+    # C:\ProgramData\Amazon\ECS\log
+    [string]$Script:ECSLogs = Initialize-Directory -Path $Script:ECSProgramData -Name  "log"
+
+    # C:\Program Files\Amazon
+    [string]$Script:AmazonProgramFiles = Initialize-Directory -Path $ENV:ProgramFiles -Name "Amazon"
+    # C:\Program Files\Amazon\ECS
+    [string]$Script:ECSProgramFiles = Initialize-Directory -Path $Script:AmazonProgramFiles -Name  "ECS"
+
+    if (-not (Test-Path $Path)) {
+        Throw "The destination path provided does not exist: $($Path)"
+        return
+    }
+}
+Process {
+    #
+    # Service Config CONSTANTS
+    #
+    [int]$SERVICE_FAILURE_RESTART_DELAY_RESET = 300

```

Comment:
nit: I'd prefer a suffix here denoting the units.  `SERVICE_FAILURE_RESTART_DELAY_RESET` is in seconds, while the other ones are in milliseconds.

Maybe:
`SERVICE_FAILURE_RESTART_DELAY_RESET_SEC`
`SERVICE_FAILURE_FIRST_DELAY_MS`
etc

Reply:
Agreed. fixing and adding to this commit